src/hooks/useEscrowContractUsers.ts

```diff
@@ -0,0 +1,48 @@
+import { createOwner, TitleEscrowOwner } from "@govtechsg/oa-token";
+
+import { getLogger } from "../utils/logger";
+
+import { useEffect, useState } from "react";
+import { useEthereumTransactionState, TransactionStateStatus, StateType } from "./useEthereumTransactionState";
+import { useWeb3Provider } from "./useWeb3Provider";
+const { trace, error } = getLogger("hooks:useEscrowContractUsers");
+
+type UseEscrowContractUsers = { state: StateType; holderAddress: string; beneficiaryAddress: string };
+
+export const useEscrowContractUsers = ({
+  escrowContractAddress
+}: {
+  escrowContractAddress: string;
+}): UseEscrowContractUsers => {
+  trace(`escrow contract address ${escrowContractAddress}`);
+  const [holderAddress, setHolderAddress] = useState("");
+  const [beneficiaryAddress, setBeneficiaryAddress] = useState("");
+  const { web3Provider } = useWeb3Provider();
+  const [state, dispatch] = useEthereumTransactionState();
+
+  useEffect(() => {
+    async function fetchEscrowContractUsers() {
+      try {
+        dispatch({ type: TransactionStateStatus.LOADING });
+        const titleEscrowInstance = (await createOwner({
+          address: escrowContractAddress,
+          web3Provider
+        })) as TitleEscrowOwner;
+        if (!titleEscrowInstance?.isTitleEscrow?.()) throw new Error("Address is not an Escrow contract");
```

Comment:
why using optional chaining everywhere ?

Reply:
not sure if the instance will have the isTitleEscrow method.

Reply:
then check what typescript say ?

Reply:
removed chaining

Reply:
as isTitleEscrowMethods is present in all the classes as RJ said.